Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp

```diff
@@ -448,15 +453,27 @@ bool MediaPlayerPrivateGStreamerBase::handleSyncMessage(GstMessage* message)
             GST_TRACE("appending init data for %s of size %" G_GSIZE_FORMAT, eventKeySystemId, mapInfo.size);
             GST_MEMDUMP("init data", reinterpret_cast<const unsigned char *>(mapInfo.data), mapInfo.size);
             concatenatedInitDataChunks.append(mapInfo.data, mapInfo.size);
+#if USE(OPENCDM)
+            initDataChunks.append(mapInfo.data, mapInfo.size);
+#endif
             ++concatenatedInitDataChunksNumber;
             eventKeySystemIdString = eventKeySystemId;
             if (streamEncryptionInformation.second.contains(eventKeySystemId)) {
                 GST_TRACE("considering init data handled for %s", eventKeySystemId);
                 m_handledProtectionEvents.add(GST_EVENT_SEQNUM(event.get()));
+#if USE(OPENCDM)
+                keySystemProtectionEventMap.add(eventKeySystemId, GST_EVENT_SEQNUM(event.get()));
+#endif
             }
             gst_buffer_unmap(data, &mapInfo);
         }
+#if USE(OPENCDM)
+        Ref<SharedBuffer> initData = SharedBuffer::create(WTFMove(initDataChunks));
+        if (!(isInitDataCached(initData)))
+            m_initDataCount++;
 
+        m_initDataProtectionEventsMap.append(std::make_pair(WTFMove(initData), keySystemProtectionEventMap));
```

Comment:
We will modify the m_initDataProtectionEventsMapping to make this changes

Reply:
We have renamed m_initDataProtectionEventsMap to m_initDataProtectionEventsMapping. And  logic changed to add only the protection event of selected KeySystem. Also events are appending to the same initData, now not creating new hash map. But still we cannot use m_initDataProtectionEventsMapping.size() for the purpose of m_initDataCount. Because this making issue in multi key case, i.e, after sending all events associated with an initdata, we are  clearing corresponding initData map from the m_initDataProtectionEventsMapping. So once it reached at the last initData map case, it show .size() value as 1, so it will try to get current session instead of session id corresponding to initData.